test/helpers/dotPathToHash.test.js

```diff
@@ -3,55 +3,79 @@ import { dotPathToHash } from '../../src/helpers'
 
 describe('dotPathToHash helper function', () => {
   it('creates an object from a string path', (done) => {
-    const res = dotPathToHash({ key: 'one' })
-    assert.deepEqual(res, { one: '' })
+    const { target, duplicate } = dotPathToHash({ key: 'one' })
+    assert.deepEqual(target, { one: '' })
+    assert.isFalse(duplicate)
     done()
   })
 
   it('ignores trailing separator', (done) => {
-    const res = dotPathToHash(
+    const { target } = dotPathToHash(
       { key: 'one.' },
       {},
       { separator: '.' }
     )
-    assert.deepEqual(res, { one: '' })
+    assert.deepEqual(target, { one: '' })
     done()
   })
 
   it('ignores duplicated separator', (done) => {
-    const res = dotPathToHash(
+    const { target } = dotPathToHash(
       { key: 'one..two' }
     )
-    assert.deepEqual(res, { one: { two: '' } })
+    assert.deepEqual(target, { one: { two: '' } })
     done()
   })
 
   it('handles a target hash', (done) => {
-    const res = dotPathToHash(
+    const { target, duplicate } = dotPathToHash(
       { key: 'one.two.three' },
       { one: { twenty: '' } }
     )
-    assert.deepEqual(res, { one: { two: { three: '' }, twenty: '' } })
+    assert.deepEqual(target, { one: { two: { three: '' }, twenty: '' } })
+    assert.isFalse(duplicate)
     done()
   })
 
   it('handles a `defaultValue` option', (done) => {
-    const res = dotPathToHash(
+    const { target } = dotPathToHash(
       { key: 'one' },
       {},
       { value: 'myDefaultValue' }
     )
-    assert.deepEqual(res, { one: 'myDefaultValue' })
+    assert.deepEqual(target, { one: 'myDefaultValue' })
     done()
   })
 
   it('handles a `separator` option', (done) => {
-    const res = dotPathToHash(
+    const { target } = dotPathToHash(
       { key: 'one_two_three.' },
       {},
       { separator: '_' }
     )
-    assert.deepEqual(res, { one: { two: { 'three.': '' } } })
+    assert.deepEqual(target, { one: { two: { 'three.': '' } } })
+    done()
+  })
+
+  it('detects duplicate keys with the same value', (done) => {
+    const { target, duplicate, conflict } = dotPathToHash(
+      { key: 'one.two.three' },
+      { one: { two: { three: '' } } },
+    )
+    assert.deepEqual(target, { one: { two: { three: '' } } })
+    assert.isTrue(duplicate)
```

Comment:
I find `isTrue` and `isFalse` hard to read. Would you mind using `equal(duplicate, true)`? It has a lower cognitive load since it is how we would write/read a test in regular js: `duplicate === true`. I also like reducing the API we use.

Reply:
Ok.